tests/test_grimgrains.py

```diff
@@ -0,0 +1,77 @@
+        self.assertEqual(
+            [
+                "nagaimo: 160 g, grated",
+                "green cabbage: 500 g, minced",
+                "chickpea flour: 85 g",
+                "nutritional yeast: 15 g",
+                "salt: 1.25 g",
+                "shiitake: 6",
+                "water: 320 ml",
+                "sesame oil: 10 ml",
+                "soy sauce: 60 ml",
+                "mirin: 60 ml",
+                "granulated sugar: 15 g",
+                "arrowroot starch: 15 g",
+                "ao nori: 30 g",
+                "beni shouga: to taste",
+                "scallions: 4 stalks",
+            ],
+            self.harvester_class.ingredients(),
+        )
+
+    def test_instructions(self):
+        self.assertEqual(
+            (
+                "Peel the nagaimo, then soak it in a water and vinegar solution (doing this helps to remove irritants). Dry the nagaimo, then grate 160 g (6-8cm) of nagaimo with a fine grater. Do this over a bowl, as the grated nagaimo is very slimy. Keep aside.\n"
+                "Mince 500 g (1 small head) green cabbage, keep aside.\n"
+                "In a bowl, mix 85 g (1 cup) of chickpea flour, 15 g (1/4 cup) nutritional yeast, the grated nagaimo, the minced green cabbage and 1.25g (1/4 tsp) of salt. Then, add 320 ml (1 1/4 cup) of water, or shiitake dashi. (for shiitake dashi, soak 5-6 shiitake in 320 ml of hot water for 15 minutes, or overnight in cold water.)\n"
+                "Heat a non-stick pan at high heat, add 5 ml (1 tsp) of sesame oil. If you throw some water on and it starts to sizzle, the pan is hot and you can add 1/4 of the batter. Alternatively, you can add 1/2, although this makes two very large portions.\n"
+                "Let okonomiyaki cook for 5 minutes, shaking the pan every now and then so the batter doesn't stick.\n"
+                "After 5 minutes, it's time to give the other side some grilling time. Put a plate on top of the pancake, keep your hand on the plate and flip the pan so that the pancake ends up cooked side up on the plate. Then, slide the pancake back into the pan (cooked side up). You can also just flip it with a spatula, but we rather like the plate method :).\n"
+                "Cook for another 5 minutes, then slide onto a plate, repeat process for the rest of the batter.\n"
+                "In a small bowl, mix 60 ml (4 tbsp) of soy sauce, 60 ml (4 tbsp) of mirin, 15 g (1 tbsp) of sugar and 15 g (1 tbsp) of arrowroot starch. Stir well.\n"
+                "Heat a pan at high heat, when hot, add sauce and cook for 2-3 minutes until it thickens. Then, divide onto your okonomiyaki.\n"
+                "First, add about 15 g (1 tbsp) (per okonomiyaki) of aonori on top of the sauce.\n"
+                "Then, add some beni shouga (pickled red ginger).\n"
+                "Finally, top off with some finely chopped scallions."
+            ),
+            self.harvester_class.instructions(),
+        )
```

Comment:
Personal preference, not a blocker for merge: I like to remove the end-of-instruction newlines from the expected instructions of the test case.  There's an example of me using [this messy-ish pattern to do that](https://github.com/hhursev/recipe-scrapers/pull/854/commits/8f1fcd149ee62e4e4bb1dd3fc1d21ef0f6c21b88).

Reply:
I was thinking about just using a multiline string but it looked kinda messy unless I removed the indentation, could it still be worth it?

Something like:

```
example = """Lorem ipsum dolor sit amet, 
consectetur adipiscing elit. Nam tempor in nisi quis 
convallis."""
```

Or, removing indentations:

```
from textwrap import dedent
example = dedent(
    """\
        Lorem ipsum dolor sit amet, 
        consectetur adipiscing elit. Nam tempor in nisi quis 
        convallis.\
    """
)
```

Reply:
I've been contributing to this codebase on-and-off for perhaps four years now, and I still haven't figured out a good way to represent the instructions as fixture data :|  But I frequently fail to consider alternatives.

So I wouldn't worry about it too much.

To me, neither the `dedent` nor multiline string approaches there feel neat enough.

Reply:
(the reverse side of that statement is: if you _do_ figure out a good representation for these, that'll be an immense relief)

Reply:
I see, I'll just change the test to the join+list

Reply:
Thank you.  I do think the in-parenthesis-multiline string approach is quite good too.  The main downside is that it leaves some ambiguity about whether a newline is within an instruction, or instead (the usual case) a delimiter between two instructions.  And it means that the string contents include some repetition, I suppose.